pages/docs/reference/js-ir-compiler.md

```diff
@@ -0,0 +1,65 @@
+---
+type: doc
+layout: reference
+category: "JavaScript"
+title: "Using the experimental IR compiler"
+---
+# Using the experimental Kotlin/JS IR compiler
+
+The (currently still experimental) Kotlin/JS IR compiler backend is the main focus of innovation around Kotlin/JS paves the way forward for the technology. Its goals are to improve on pain points that the previous compiler had, in areas of developer experience, size of generated code (through dead code elimination), and to improve interoperability with the JavaScript and TypeScript ecosystems, to name just a few.
```

Comment:
Can we somehow introduce the IR compiler backend - what is it in general?
Or at least explain what is IR.
Maybe take some parts of your blog posts?

Reply:
I've tried to include some high-level information on the topic, while trying to avoid to dive to deep into the insides of the compiler (I'm not sure how relevant it is to the _user_ of the compiler). Let me know what you think, otherwise happy to give it another iteration.

Reply:
Great, this is the explanation I was missing here. Just the explanation of what is IR, how it's used, and how we can benefit from it.